realm/realm-library/src/androidTest/java/io/realm/DynamicRealmObjectTests.java

```diff
@@ -1924,7 +1924,8 @@ public void getNullableFields() {
                 case STRING_TO_DECIMAL128_MAP:
                 case STRING_TO_LINK_MAP:
                 case STRING_SET:
-                    // TODO: fix this when adding support for dictionaries
+                case INTEGER_SET:
+                    // TODO: fix this when adding support for dictionaries and sets
```

Comment:
Same as below. Additionally, these switches are tricky because they assume support for that very type is fully integrated in the SDK, which is not the SDK :-( I will add a reference here to the tickets so that it doesn't vanish.

Reply:
On top of that, some switches perform checks assuming that things would throw and some don't, it's just really confusing 🤯